src/Storages/Statistics/StatisticsMinMax.cpp

```diff
@@ -35,6 +36,18 @@ void StatisticsMinMax::update(const ColumnPtr & column)
     row_count += column->size();
 }
 
+void StatisticsMinMax::merge(const SingleStatisticsPtr & other)
```

Comment:
For my understanding: Is it correct to say that the new `merge` functions and their 4 implementations in the existing statistics types are there for optimization purposes, i.e. instead of updating the statistics with incoming row values, they are updated with existing statistics (which may be faster)?

I think this is a good optimization but we should rather keep it out of this refactoring and push it separately.

Reply:
`merge ` is only used when we want to merge 2 statistics. It is used when query hits 2 parts and we merge the statistics.